syntax = "proto3";

message Point {
 float x =1;
 float y =2;
}
message MapPartitionRequest {
  int32 start = 1;
  int32 end = 2;
  int32 numMappers=3;
  int32 numReducers =4;
  repeated Point centroids =5;
}


message MapPartitionResponse {
}

message InvokeReducerRequest {
}

message InvokeReducerResponse {
}

message GetInputDataRequest {
}

message GetInputDataResponse {
}

service MasterMapper {
  rpc Map (MapPartitionRequest) returns (MapPartitionResponse) {}
  rpc InvokeReducer (InvokeReducerRequest) returns (InvokeReducerResponse) {}
  rpc GetInputData (GetInputDataRequest) returns (GetInputDataResponse) {}
}
